src/key.rs

```diff
@@ -364,14 +365,26 @@ impl PublicKey {
     /// the result would be the point at infinity, i.e. we are adding this point
     /// to its own negation
     pub fn combine(&self, other: &PublicKey) -> Result<PublicKey, Error> {
+        PublicKey::combine_keys(&[self, other])
+    }
+
+    /// Adds the keys in the provided slice together, returning the sum. Returns
+    /// an error if the result would be the point at infinity, i.e. we are adding
+    /// a point to its own negation
+    pub fn combine_keys(keys: &[&PublicKey]) -> Result<PublicKey, Error> {
```

Comment:
Post-merge note: `PublicKey` is a copy type and I am not sure why we need to have it here by reference

Reply:
IIRC the reason is because we need a `&[&PublicKey]` to transmute. It would also be possible to take a `&[PublicKey]` and do the conversion internally but I guess I preferred to keep the library code minimal. I don't have a strong opinion if people want to change it though.

Reply:
Yeah the C code takes pointer to a list of pointers, so we'd have to allocate a vector of pointers and we try to avoid allocations where possible